shared/src/vyxal/MiscHelpers.scala

```diff
@@ -160,6 +169,38 @@ object MiscHelpers:
       i += 1
     VList.from(result.result())
 
+  def setObjectMember(obj: VObject, name: String, value: VAny)(using
+      ctx: Context
+  ): VObject =
+    val (visibility, _) = obj.fields
+      .getOrElse(name, throw FieldNotFoundException(obj.className, name))
+
+    val objName = obj.className
+    val newObj = obj.copy()
+    visibility match
+      case Visibility.Public => newObj.fields(name) = (visibility, value)
```

Comment:
> [!warning]
> `.copy()` only does a shallow copy, not a deep copy. This means that `newObj.fields` is the same object as the old `obj.fields` (unless you've overridden it somewhere, but I didn't see that). Make `fields` an immutable map, and create a new `fields` map here instead of setting a single value. Also, we need to add tests to make sure setting a field only returns a new object but doesn't mutate existing objects

That didn't need an admonition but they're so cool